### PR TITLE
lint project before commit prevents fails in CI

### LINKED ISSUE #117: Linting before commit
It is annoying to see linting error after changes have been made, integrating it into existing githook is nice way to prevent it, I think.

### PR DESCRIPTION
**Pull request by mrtrkmn**
fixes #117

### CODE REVIEW

**Comment by mrtrkmn** [comment]
Note that the failing test case is NOT about  the changes in this pull request.

File: .githooks/pre-commit.linux.sample [hunk 1]
```diff
@@ -1,3 +1,25 @@
...
+                                          grep -xF "$package_name" | grep autopep8)" != "autopep8" ]
+then
+    echo 'autopep8 is NOT installed, linting test may fail on CI ... '
+    echo 'consider to install autopep8, you may use following commands: '
+    echo 'Debian: [ sudo apt-get install -y python-autopep8 ] '
+    echo 'MacOS: [ brew install autopep8 ]'
+    echo 'You may consider to install it into virtual environment of your project:'
+    echo 'source venv/bin/activate'
+    echo 'pip3 install autopep8'
+    echo 'autopep8 should be available in your system, to do not face with linting problem.'
```
Comment (mrtrkmn) [inline .githooks/pre-commit.linux.sample:15]: This part is written to inform person to do what.

File: .githooks/pre-commit.linux.sample [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+    exit 1
+else
+    echo 'Linting...'
+    echo 'Going to root directory of the project'
+    cd ../richkit
+    autopep8 --in-place --recursive --max-line-length=100 --exclude docs/source/conf.py,venv,__pycache__,old,build,dist .
```
Comment (mrtrkmn) [inline .githooks/pre-commit.linux.sample:21]: Linting ....

**Review by kidmose — COMMENTED** [review]
:+1:  overall, but one point I'd like to have clarified;
This seems to (potentially) modify the working tree with `autopep8`, but after committer decides what is to be commit and after creating the commit message.
Do we understand where the changes by `autopep8` will go? 
It seems to me that, as long as `autopep8` is available, the staged changes will be committed whether there are pep8 corrections to be made or not, and any changes by `autopep8` will be left in the working tree. Is that correctly understood and is that as intended?

**Review by gianmarcomennecozzi — APPROVED** [review]

**Comment by mrtrkmn** [comment]
> 👍 overall, but one point I'd like to have clarified;
> This seems to (potentially) modify the working tree with `autopep8`, but after committer decides what is to be commit and after creating the commit message.
> Do we understand where the changes by `autopep8` will go?
> It seems to me that, as long as `autopep8` is available, the staged changes will be committed whether there are pep8 corrections to be made or not, and any changes by `autopep8` will be left in the working tree. Is that correctly understood and is that as intended?

Yes good point to mention, it is annoying to review changes which are caused by `autopep8` in one commit, however it cannot be seen multiple times. Since other parts of the project will be already linted and just the part where changes made will be effected by this hook, . So, if the code base already linted by `autopep8`, the parts other than the actual changes will not be affected. 

I hope, I have understood your concerns correctly and responded accordingly.

**Comment by kidmose** [comment]
> I hope, I have understood your concerns correctly and responded accordingly.

I'm not sure, but it's minor so let's let it fly :)

**Review by kidmose — APPROVED** [review]

**Comment by mrtrkmn** [comment]
> I'm not sure, but it's minor so let's let it fly :)

I mean, once everything is linted, there will be no more linting other than changed files for a commit.

**Comment by mrtrkmn** [comment]
I will merge at the end, this way `develop` will be linted and include latest changes.

**Pull request merged**